api.go

```diff
@@ -0,0 +1,219 @@
+package main
+
+/*
+ Portions of this file were derived from Dave Cheney's httpstat:
+ https://github.com/davecheney/httpstat
+
+ His code is licensed as follows:
+
+     MIT License
+
+    Copyright (c) 2016 Dave Cheney
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in all
+    copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+
+*/
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptrace"
+	"net/textproto"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+)
+
+var placeholderRegex = regexp.MustCompile(`{{ *[^}}]* *}}`)
```

Comment:
You probably don't want to go down this road but a future endeavor would be to incorporate the [template](https://golang.org/pkg/text/template/) library to make templating these variables easier.